pkg/registry/apis/provisioning/test.go

```diff
@@ -70,6 +70,7 @@ func (s *testConnector) Connect(ctx context.Context, name string, opts runtime.O
 					return
 				}
 
+				// TODO: remove this hack and it's not needed needed anymore as we validate OnCreate
```

Comment:
maybe re-phrase as 
```suggestion
				// TODO: Explore how to better support synchronous validation for the UI (and likely remove this hack)
```
or similar.   The synchronous healthcheck OnCreate/Update should likely be removed... but that is a different discussion